sky/clouds/gcp.py

```diff
@@ -470,27 +488,10 @@ def accelerator_in_region_or_zone(self,
 
     @classmethod
     def get_project_id(cls, dryrun: bool = False) -> str:
-        # TODO(zhwu): change the project id fetching with the following command
-        # `gcloud info --format='value(config.project)'`
         if dryrun:
             return 'dryrun-project-id'
-        if 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
-            gcp_credential_path = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
-        else:
-            gcp_credential_path = DEFAULT_GCP_APPLICATION_CREDENTIAL_PATH
-        if not os.path.exists(gcp_credential_path):
-            with ux_utils.print_exception_no_traceback():
-                raise FileNotFoundError(
-                    f'No GCP credentials found at '
-                    f'{gcp_credential_path}. Please set the '
-                    f'GOOGLE_APPLICATION_CREDENTIALS '
-                    f'environment variable to point to '
-                    f'the path of your credentials file.')
-
-        with open(gcp_credential_path, 'r') as fp:
-            gcp_credentials = json.load(fp)
-        project_id = gcp_credentials.get('quota_project_id',
-                                         None) or gcp_credentials['project_id']
```

Comment:
The `auth.default()` is official method that returns the credential and project id, which retrieves the project id just as we did before. I think it should be cleaner to use the official method instead of doing it by our own?
https://google-auth.readthedocs.io/en/master/reference/google.auth.html
https://google-auth.readthedocs.io/en/master/_modules/google/auth/_default.html#default